persist/sqlite/contracts.go

```diff
@@ -864,11 +864,52 @@ raw_revision, host_sig, renter_sig, confirmed_revision_number, contract_status,
 	return
 }
 
+// v2ContractExists is a helper that checks if a contract already exists.
+//
+// If the contract exists, and its contract_status is rejected the
+// existing contract will be deleted so the new contract can be inserted
+// and returns (nil).
+//
+// If the contract exists, and its contract_status is not rejected,
+// [contracts.ErrContractExists] will be returned.
+//
+// If the contract does not exist, it returns (nil).
+
+func v2ContractExists(tx *txn, contractID types.FileContractID) error {
```

Comment:
`deleteRejectedV2Contract`? The current name doesn't really indicate that a deletion is happening.